src/Microsoft.AspNetCore.Mvc.Razor.Precompilation/Internal/PrecompileCommandBase.cs

```diff
@@ -0,0 +1,184 @@
+                "Type with Configure method",
+                CommandOptionType.SingleValue);
+
+            ContentRootOption = app.Option(
+                "--content-root",
+                "The application's content root.",
+                CommandOptionType.SingleValue);
+
+            GeneratePdbOption = app.Option(
+                "--generate-pdbs",
+                "Generate pdbs for views.",
+                CommandOptionType.NoValue);
+
+            app.OnExecute((Func<int>)Execute);
+        }
+
+        protected int Execute()
+        {
+            if (!ParseArguments())
+            {
+                return 1;
+            }
+
+            return ExecuteCore();
+        }
+
+        protected abstract int ExecuteCore();
+
+        private bool ParseArguments()
+        {
+            if (!string.IsNullOrEmpty(ProjectArgument.Value))
+            {
+                ProjectPath = Path.GetFullPath(ProjectArgument.Value);
+                if (string.Equals(Path.GetFileName(ProjectPath), "project.json", StringComparison.OrdinalIgnoreCase))
+                {
+                    ProjectPath = Path.GetDirectoryName(ProjectPath);
+                }
+
+                if (!Directory.Exists(ProjectPath))
+                {
+                    Console.Error.WriteLine($"Error: Could not find directory {ProjectPath}");
+                    return false;
+                }
+            }
+            else
+            {
+                ProjectPath = Directory.GetCurrentDirectory();
+            }
+
+            if (!FrameworkOption.HasValue())
+            {
+                var workspace = new BuildWorkspace(ProjectReaderSettings.ReadFromEnvironment());
+                var projectContexts = workspace.GetProjectContextCollection(ProjectPath)
+                    .FrameworkOnlyContexts
+                    .ToList();
+
+                if (projectContexts.Count == 0)
+                {
+                    Console.Error.WriteLine($"Error: Project at {ProjectPath} does not have any specified frameworks.");
+                    return false;
+                }
+                else if (projectContexts.Count > 1)
+                {
+                    Console.Error.WriteLine($"Error: Project at {ProjectPath} supports multiple framework. " +
+                        $"Specify a framework using {FrameworkOption.Template}.");
+                    return false;
+                }
+
+                TargetFramework = projectContexts[0].TargetFramework;
+            }
+            else
+            {
+                TargetFramework = NuGetFramework.Parse(FrameworkOption.Value());
+            }
+
+            if (ConfigurationOption.HasValue())
+            {
+                Configuration = ConfigurationOption.Value();
+            }
+            else
+            {
+                Configuration = "Debug";
+            }
+
+            return true;
+        }
+
+        protected string GetBuildOutputPath()
```

Comment:
You could pass the inside man this information so it didn't need to re-calculate it couldnt you?